testr/testr.go

```diff
@@ -67,6 +87,44 @@ type Underlier interface {
 	GetUnderlying() *testing.T
 }
 
+// UnderlierInterface exposes access to the underlying TestingT instance. Since
+// callers only have a logr.Logger, they have to know which
+// implementation is in use, so this interface is less of an
+// abstraction and more of a way to test type conversion.
+type UnderlierInterface interface {
+	GetUnderlying() TestingT
+}
+
+func logInfo(
```

Comment:
Also please add comments on these funcs to explain why they exist

Reply:
I added some comments to the new functions. They are here to prevent code duplication between `testlogger` and `testloggerInterface`.